solr/core/src/java/org/apache/solr/storage/CompressingDirectory.java

```diff
@@ -92,6 +92,15 @@ static OpenOption getDirectOpenOption() {
   private final boolean useAsyncIO;
   private final boolean useDirectIO;
 
+  public CompressingDirectory(
```

Comment:
Excuse me if I missed it but I'm not seeing this new constructor used anywhere, is this necessary now or for something in the future?

Reply:
Added a public ctor here in order to make `NodeLevelTeeDirectoryState` class public (for external use, e.g. in tests and transfer scripts), without having to make the internal fields accessible.

This is a good callout though, let me see if I can adjust this to make it a little more clear how this is supposed to be used.

Reply:
Updated usages to make it clear that the preferred/public ctor is the one that takes `NodeLevelTeeDirectoryState` arg. This will hopefully make it a little less unusual-looking!